### PR TITLE
SNAKE AI (PASSES TURING TEST) (REACTIVE COWBOY BOOT LOGIC) 😱😱😱😱😱

### PR DESCRIPTION
**Pull request by tralezab**
## About The Pull Request

makes a datum ai controller for snakes, they are now very reclusive and do not like being touched. they will seek out boots to hide in, mice to hunt, and enemies to exact revenge on. feeding mice will make the snake like you, allowing you to drag it around and touch it without it getting mad at you, while also doing things that annoy it defriending instead of making it hostile.

## Why It's Good For The Game

deletes retaliate/poison, hate that path man

## Changelog
:cl:
add: Snake AI, they now are way more reclusive unless they are fed mice. Keep your boot away from them!
/:cl:

### CODE REVIEW

File: code/datums/ai/snake/snake_behaviors.dm [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+	walk(living_pawn, 0)
+	controller.blackboard[BB_SNAKE_CURRENT_RETREAT_TARGET] = null
+
+/datum/ai_behavior/snake_attack_mob
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_MOVE_AND_PERFORM //performs to increase frustration
+
+/datum/ai_behavior/snake_attack_mob/perform(delta_time, datum/ai_controller/controller)
+	. = ..()
+
+	var/mob/living/target = controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET]
```
Comment (MrDoomBringer) [inline code/datums/ai/snake/snake_behaviors.dm:36]: monki

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]

File: code/datums/ai/snake/snake_behaviors.dm (hunk 1 above)
Comment (MrDoomBringer) [reply code/datums/ai/snake/snake_behaviors.dm:36]: theres a few other occurrences of what looks like monkey defines when it should be snake defines, but i may just be misunderstanding something

Comment (tralezab) [reply code/datums/ai/snake/snake_behaviors.dm:36]: FRIDGE! Thanks for catching that

**Comment by tralezab** [comment]
Floyd and I are talking about this, needs to happen after mob AI gets generic datumized

**Pull request closed without merging**